mlflow/server/js/src/common/utils/Utils.js

```diff
@@ -236,29 +236,23 @@ class Utils {
     return withoutTrailingSlash;
   }
 
-  static getGitHubRegex() {
-    return /[@/]github.com[:/]([^/.]+)\/([^/#]+)#?(.*)/;
-  }
-
-  static getGitLabRegex() {
-    return /[@/]gitlab.com[:/]([^/.]+)\/([^/#]+)#?(.*)/;
+  static getGitRegex() {
+    return /(.*?[@/][^?]*git.*?)[:/]([^#]+)(?:#(.*))?/;
```

Comment:
Could you comment some example strings that the regex can match ?

Reply:
I add comment now 🌈

Reply:
I suggest split regex for each case matching string,

e.g.,

case 1: `github.com` / `gitlab.com` ,  we should match the whole string of `github` / `gitlab`,

case 2: custom git repo like `https://git.XXX. ...)`, 

This is for better code readability, more strict matching rule, and easier code extensibility.

Reply:
Thank you for reply!

I added a new regex and left the old ones 🌈

Reply:
@WeichenXu123 

Hi.
Can you check this?